apps/engine/lib/engine/release_tasks/contract.ex

```diff
@@ -31,34 +32,53 @@ defmodule Engine.ReleaseTasks.Contract do
     default_url = config |> Keyword.fetch!(:ethereumex) |> Keyword.fetch!(:url)
     rpc_url = "ETHEREUM_RPC_URL" |> get_env() |> Validators.url("ETHEREUM_RPC_URL", default_url)
 
-    [
-      payment_exit_game,
-      eth_vault,
-      erc20_vault,
-      min_exit_period_seconds,
-      contract_semver,
-      child_block_interval,
-      contract_deployment_height
-    ] = external_data(plasma_framework, tx_hash, rpc_url)
-
-    Config.Reader.merge(config,
-      engine: [
-        rpc_url: rpc_url,
-        authority_address: authority_address,
-        plasma_framework: plasma_framework,
-        eth_vault: eth_vault,
-        erc20_vault: erc20_vault,
-        payment_exit_game: payment_exit_game,
-        min_exit_period_seconds: min_exit_period_seconds,
-        contract_semver: contract_semver,
-        child_block_interval: child_block_interval,
-        contract_deployment_height: contract_deployment_height
-      ]
-    )
+    contracts_config = get_contracts_config(plasma_framework, tx_hash, rpc_url)
+
+    engine_config =
+      Keyword.merge(
+        [
+          rpc_url: rpc_url,
+          authority_address: authority_address,
+          plasma_framework: plasma_framework
+        ],
+        contracts_config
+      )
+
+    Config.Reader.merge(config, engine: engine_config)
+  end
+
+  defp get_contracts_config(plasma_framework, tx_hash, rpc_url) do
+    case get_contracts_config_from_db() do
+      nil ->
+        config = get_config_from_root_chain(plasma_framework, tx_hash, rpc_url)
+        :ok = store_contracts_config_in_db(config)
+        config
```

Comment:
your parent function is called `get_contracts_config `... *GET*, and then you write in the DB in their body, split this out.